src/libraries/System.Text.Json/src/System/Text/Json/Node/JsonObject.List.cs

```diff
@@ -0,0 +1,386 @@
+                if (ReferenceEquals(item.Value, node.Value) && GetStringComparer().Compare(item.Key, node.Key) == 0)
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private bool ContainsNode(JsonNode? node)
+        {
+            CreateList();
+
+            foreach (JsonNode? item in GetValueCollection(this))
+            {
+                if (ReferenceEquals(item, node))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private KeyValuePair<string, JsonNode?>? FindNode(JsonNode? node)
+        {
+            CreateList();
+
+            foreach (KeyValuePair<string, JsonNode?> item in this)
+            {
+                if (ReferenceEquals(item.Value, node))
+                {
+                    return item;
+                }
+            }
+
+            return null;
+        }
+
+        private bool ContainsNode(string propertyName)
+        {
+            CreateList();
+
+            if (_dictionary != null)
+            {
+                return _dictionary.ContainsKey(propertyName);
+            }
+
+            foreach (string item in GetKeyCollection(this))
+            {
+                if (GetStringComparer().Compare(item, propertyName) == 0)
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private bool TryRemoveNode(string propertyName, out JsonNode? existing)
+        {
+            CreateList();
+            Debug.Assert(_list != null);
+
+            if (_dictionary != null)
+            {
+                if (!_dictionary.TryGetValue(propertyName, out existing))
+                {
+                    return false;
+                }
+
+                bool success = _dictionary.Remove(propertyName);
+                Debug.Assert(success);
+            }
+
+            for (int i = 0; i < _list.Count; i++)
+            {
+                KeyValuePair<string, JsonNode?> current = _list[i];
+                if (GetStringComparer().Compare(propertyName, current.Key) == 0)
+                {
+                    _list.RemoveAt(i);
+                    existing = current.Value;
+                    DetachParent(existing);
+                    return true;
+                }
+            }
+
+            existing = null;
+            return false;
+        }
+
+        private bool TryFindNode(string propertyName, out JsonNode? property)
+        {
+            CreateList();
+            Debug.Assert(_list != null);
+
+            if (propertyName == _lastKey)
+            {
+                // Optimize for repeating sections in code:
+                // obj.Foo.Bar.FirstProperty = value1;
+                // obj.Foo.Bar.SecondProperty = value2;
+                property = _lastValue;
+                return true;
+            }
+
+            bool success;
```

Comment:
Nit: this is only used inside of the `if (_dictionary != null)` so it could be moved down there.

Reply:
Thanks